src/components/tools/geometry-tool/use-movable-line-dialog.tsx

```diff
@@ -0,0 +1,132 @@
+
+interface IContentProps {
+  slope: string;
+  setSlope: React.Dispatch<React.SetStateAction<string>>;
+  intercept: string;
+  setIntercept: React.Dispatch<React.SetStateAction<string>>;
+  errorMessage: string;
+}
+const Content: React.FC<IContentProps> = ({ slope, setSlope, intercept, setIntercept, errorMessage }) => {
+  return (
+    <div className="movable-line-dialog-content">
+      <LineOption
+        display="Slope:"
+        id="slopeInput"
+        defaultValue={slope}
+        setValue={setSlope}
+      />
+      <LineOption
+        display="y-intercept:"
+        id="yinterceptInput"
+        defaultValue={intercept}
+        setValue={setIntercept}
+      />
+      <div className="dialog-error-message">
+        {errorMessage}
+      </div>
+    </div>
+  );
+};
+
+interface IProps {
+  line: JXG.Line;
+  onAccept: (line: JXG.Line, point1: [number, number], point2: [number, number]) => void;
+  onClose: () => void;
+}
+export const useMovableLineDialog = ({ line, onAccept, onClose }: IProps) => {
+  const [slope, setSlope] = useState(`${round(line.getSlope(), 1)}`);
+  const [intercept, setIntercept] = useState(`${round(line.getRise(), 1)}`);
+
+  const fSlope = parseFloat(slope);
+  const fIntercept = parseFloat(intercept);
+  const errorMessage = (!isFinite(fSlope) || !isFinite(fIntercept))
+                        ? "Please enter a valid number for slope and intercept."
+                        : getBoundingBoxIntersections(fSlope, fIntercept, line.board).length < 2
+                          ? "Please change the graph axes scale to make the line visible, and try again."
+                          : "";
+
+  const getLineControlPoints = () => {
+    if (isFinite(fSlope) && isFinite(fIntercept)) {
+      const board = line.board;
+      const intersections = getBoundingBoxIntersections(fSlope, fIntercept, board);
+      if (!intersections.length) return undefined;
+      let point1: [number, number];
+      let point2: [number, number];
+      if (board.hasPoint(0, fIntercept)) {
+        point1 = [0, fIntercept];
+        const maxX = intersections[1][0];
+        const middleX = maxX / 2;
+        point2 = [middleX, solveForY(fSlope, fIntercept, middleX)];
+      } else {
+        const minX = intersections[0][0];
+        const maxX = intersections[1][0];
+        const lineWidth = maxX - minX;
+        const thirdLineWidth = lineWidth / 3;
+        const x1 = minX + thirdLineWidth;
+        const x2 = minX + thirdLineWidth * 2;
+        point1 = [x1, solveForY(fSlope, fIntercept, x1)];
+        point2 = [x2, solveForY(fSlope, fIntercept, x2)];
+      }
+      return [point1, point2];
+    } else {
+      return undefined;
+    }
+  };
+
+  const handleClick = () => {
+    const points = getLineControlPoints();
+    if (points) {
+      onAccept(line, points[0], points[1]);
+    } else {
+      onClose();
+    }
+  };
+
+  const [showModal, hideModal] = useCustomModal({
+    Icon: MovableLineSvg,
+    title: "Edit Line",
```

Comment:
I realize this was the way it was before, but `Movable Line` would be a better title.